### DISCUSSION TITLE
Upgraded to 3.1.3 without reading release notes--oops.

### DISCUSSION THREAD

**Discussion by danb35** [osy86/HaC-Mini#964] [Q&A] [answered]
My HaC Mini (a NUC8i7HNK) had been mostly-happily running Catalina (except for the occasional unexpected reboot) using hac-mini 2.17.  But prompted by those reboots, I went ahead and upgraded to 3.1.3 without following the 3.0.1 release notes and resetting the NVRAM beforehand.  Oops.

After doing the update, of course my system didn't boot any more.  So, now looking at the release notes, I see that one of the fixes is to boot into the internal EFI shell and run `fs0:\EFI\OC\OpenCore.efi`.  I ran that, whereupon I was greeted by a black screen--nothing at all there.  So I rebooted, and entered the boot picker, where the only option was "Reset NVRAM", which I chose.  The system rebooted into the same black screen; subsequent reboots into the boot picker again show only Reset NVRAM as an option.

The last option on the list in the release notes was to reset the BIOS, so I entered the BIOS menu and reset it to defaults, and then made the recommended changes (USB power, Thunderbolt, Secure Boot).  No change.  I then tried updating the BIOS (latest available, from May of this year), resetting again, once again making those changes.  Again, no change.

Getting a little concerned now.  Downloaded Catalina on my MacBook Pro, made a fresh USB of it using `sudo "/Applications/Install macOS Catalina.app/Contents/Resources/createinstallmedia" --volume /Volumes/Installer`, and installed 3.1.3 on that (checking the box for the boot menu).  Take it to my NUC, and it will boot; the boot menu options are EFI, Install macOS Catalina, and Reset NVRAM.  I don't see an option to boot from the existing internal SSD that already has Catalina on it.

If I launch `Install macOS Catalina`, and then go under the Apple menu to `Select Startup Disk`, the internal SSD is shown.  If I go to Disk Utility, it's there as well.  But it isn't showing up as a bootable device.  What should my next steps be?

**Comment by osy** [comment]
Do you have one ssd installed or two? Can you see it in the startup utility in recovery mode?

**Comment by danb35** [comment]
I have only one SSD installed.  If by "recovery mode" you mean when I boot into "install macOS Catalina" from the USB, and the startup utility is what comes up when I select "Startup Disk" from the Apple menu, yes, it comes up there.

Edit: I'm marking this as the answer, as I can't mark a reply to this comment as the answer.  The solution is found in the comment by @x2358473 noting an apparent incompatibility between 3.1.3 and Catalina, and recommending use of 3.0.1.

**Reply by osy** [reply]
And when you select it from there, it still doesn’t boot?

**Reply by danb35** [reply]
When I select it from there and click Restart, it asks for confirmation.  When I click Restart again, it restarts and takes me back to the boot menu from the USB, where the options are EFI, Install macOS Catalina, and Reset NVRAM.

**Reply by osy** [reply]
Hmm a few people have run into this but I can’t seem to reproduce it. If you go into EFI shell without the USB drive inserted, at the top should be a list of drives it recognizes. Can you take a picture of that?

**Reply by danb35** [reply]
Sure, here you go.

The only storage devices are the single internal SSD, and the USB stick.  I see that it indicates another "BlockDevice", a "Removable HardDisk", and a "Removable BlockDevice" apart from those two; I don't know what those would be.  There was a USB hub attached with a variety of non-storage devices attached, but unplugging it didn't change the output here.

![21-11-22 13-39-38 2559](https://user-images.githubusercontent.com/12121759/142917589-f7b1dc5b-82a9-40d0-ac29-c4024b19c4b9.jpg)

**Reply by osy** [reply]
Hmm fs0 does appear, can you boot from fs0:EFI/OC/OpenCore.efi? Also try unplugging all other USB while troubleshooting.

**Reply by danb35** [reply]
Unplugged the installer USB, so the only thing plugged into USB is my keyboard.  Booted back into the EFI shell and ran `fs0:\EFI\OC\OpenCore.efi`.  The system is now showing a completely black screen.

**Reply by osy** [reply]
Oh can you spam the "Alt" key to get into the boot picker menu?

So as soon as you type in ...OpenCore.efi and press enter start spamming Alt as fast as possible.

**Reply by danb35** [reply]
I do get the boot picker menu when I do that; the only entry is Reset NVRAM.

**Reply by osy** [reply]
This is weird. Have you tried re-installing macos?

**Reply by danb35** [reply]
I think that's the next step.

**Reply by x2358473** [reply]
目前Catalina 只能使用3.0.1。 从3.1.2开始在Catalina上就无法显示系统SSD的的引导选项。只有EFI、install mac osx 、重置NVRAM这三个选项。只能等大神解决。我也是喜欢Catalina。所以我目前只能停留在3.0.1的 Hac Mini.
Currently Catalina can only use 3.0.1. Starting from 3.1.2, the boot options of the system SSD cannot be displayed on Catalina. There are only three options: EFI, install mac osx, and reset NVRAM. Can only wait for the great god to solve it. I also like Catalina. So I can only stay at Hac Mini 3.0.1 at the moment.

**Reply by danb35** [reply]
I think this is the answer.  I created a fresh Catalina installer USB, installed 3.0.1 on it, reinstalled macOS, and now I'm up and running.  It appears there's some incompatibility between 3.1.3 and Catalina.

**Reply by Medizinmann-MD** [reply]
Maybe this should be mentioned in the release notes!

**Reply by goodbest** [reply]
@danb35 @x2358473 @Przeblysk  @fucongwanglucid @bosbode Do you enable the "SecureBoot" option for Hac-mini installer in Catalina? (I'm not mean the secure boot in BIOS)
If you do enable that, more config is needed in Hac Mini 3.1.x for Catalina. See link below.
If you do not enable SecureBoot in the installer, then report back first.

cc @osy 
More detail in OC 0.7.2 (used in Hac-mini 3.1.x)
https://www.reddit.com/r/hackintosh/comments/owkrcn/psa_opencore_072_will_require_most_to_jump/
https://dortania.github.io/OpenCore-Post-Install/universal/security/applesecureboot.html#dmgloading

`First off, the default minimum date and version required for APFS drivers to load has been bumped up to Big Sur. This means APFS drivers that come with High Sierra through Catalina won't load.`

**Reply by danb35** [reply]
The only setting I changed from the default was to patch the installer when I created a new USB.  I didn't change anything relating to secure boot or anything else.

**Reply by Przeblysk** [reply]
I'm sure I didn't open SecureBoot in the installer.The default installation option is used. @goodbest

**Reply by osy** [reply]
@goodbest Somehow I missed that comment, oof

**Comment by osy** [comment, marked as answer]
This should be fixed in v3.1.4
